controllers/auth.js

```diff
@@ -0,0 +1,58 @@
+
+const User = require("../models/User");
+const bcrypt = require("bcrypt");
+
+// User Signup
+const signup = async (req, res)=>{
+    req.body = {
+        "username": "faizan",
+        "fullname": "FaizanQazi",
+        "email": "faizn@gmail.com",
+        "password": "123124243"
+    }
+    try {
+            // Checking if the user already exist with email or username
+            await User.findOne({email: req.body.email}) && res.status(500).json("User with this email already exists");
+
+            await User.findOne({username: req.body.username}) && res.status(500).json("Username  not available");
```

Comment:
You can make the field unique instead of running an additional query: https://www.mongodb.com/docs/v5.0/core/index-unique/